meteor/client/ui/SegmentContainer/withResolvedSegment.ts

```diff
@@ -331,17 +353,7 @@ export function withResolvedSegment<T extends IProps, IState = {}>(
 						nextProps.playlist.nextSegmentId === props.segmentId)) ||
 				((props.playlist.currentPartInstanceId !== nextProps.playlist.currentPartInstanceId ||
 					props.playlist.nextPartInstanceId !== nextProps.playlist.nextPartInstanceId) &&
-					data.parts &&
-					(data.parts.find(
-						(i) =>
-							i.instance._id === props.playlist.currentPartInstanceId ||
-							i.instance._id === nextProps.playlist.currentPartInstanceId
-					) ||
-						data.parts.find(
-							(i) =>
-								i.instance._id === props.playlist.nextPartInstanceId ||
-								i.instance._id === nextProps.playlist.nextPartInstanceId
-						))) ||
+					((data.parts && findNextOrCurrentPart(data.parts)) || data.segmentui?.showShelf)) ||
```

Comment:
`data.segmentui?.showShelf` doesn't seem right here. I don't remember the reason it was added, but it causes segments with Mini Shelves to update way too often.